workshopapps/todoassistant.web
# Feat(FRO-64): Implement Virtual Assistant client page

### Task : Virtual Assistant clients View with functionality
**Task Description**: consume the API thats displays all client associated with a Virtual Assiststant

**API ENDPOINTS:** 
**/va/user/:va_id**  -- returns the user associated with the VA
**va/user/profile/:user_id** -- return full details of the user when VA click "view profile"

**VIRTUAL ASSISTANT LOGIN CRED:**
email address: 2@a.com
password: secret123    (not so secret now tho)

**FIGMA LINK:**
https://www.figma.com/file/3f7fQODyHQ0PC7e2skvD8O/Ticked?node-id=4180%3A61304&t=rKTPZz64uv5zeZkg-0

**LIVE LINK:** 
https://ticked.hng.tech/virtual-assistance/clients

**HOW TO TEST:**
_step 1:_ login as a virtual assistant with the credentials above
_step 2:_ go to the client section on the side navigation(click it)
at this point the api (returns the users associated with the VA) is called,
a table is displayed with the users.

**TO SEE FULL USER DETAIL (second APi call)**
_step 1:_ click the dropdown available in the row of a user (click view profile)
_step 2:_  user details is displayed in a modal.


**_ERRORS YOU MIGHT ENCOUNTER:_**
1. Internal server error -- error 500
2. data returned by api might render in HTML/text instead of application/json 
this error only happened on the live site! **(backend dev said its a deployment issue)**

### **VIEWS**
![Screenshot 2022-12-02 075716](https://user-images.githubusercontent.com/64202931/205450156-02571044-9fe6-4f29-a332-9d3d71a04b47.png)
![Screenshot 2022-12-03 170101](https://user-images.githubusercontent.com/64202931/205450157-089d5915-7b95-4318-acd7-732b1058836c.png)
